File: src/main/java/c6h2cl2/mod/SolidXp/Item/Armor/ItemXpIronArmor.java
package c6h2cl2.mod.SolidXp.Item.Armor;

import c6h2cl2.mod.SolidXp.SolidXpRegistry;
import net.minecraft.inventory.EntityEquipmentSlot;
import net.minecraft.item.ItemArmor;

/**
 * Created by C6H2Cl2 on 2016/09/06.
 */
public class ItemXpIronArmor extends ItemArmor implements IXpArmor{
    public ItemXpIronArmor( EntityEquipmentSlot equipmentSlotIn) {
        super(SolidXpRegistry.armorMaterialXpIron,0, equipmentSlotIn);
    }

    @Override
    public float getArmorShield() {
        return 10;
    }

    @Override
    public float getShieldBonus() {
        return 10;
    }

    @Override
    public float getShieldLimit() {
        return 20;
    }
}

File: src/main/java/c6h2cl2/mod/SolidXp/Item/Armor/ItemXpIronArmor.kt
package c6h2cl2.mod.SolidXp.Item.Armor

import c6h2cl2.mod.SolidXp.SolidXpRegistry
import net.minecraft.inventory.EntityEquipmentSlot
import net.minecraft.item.ItemArmor

class ItemXpIronArmor(equipmentSlotIn: EntityEquipmentSlot) : ItemArmor(SolidXpRegistry.armorMaterialXpIron, 0, equipmentSlotIn), IXpArmor {

    override fun getArmorShield(): Float {
        return 10f
    }

    override fun getShieldBonus(): Float {
        return 10f
    }

    override fun getShieldLimit(): Float {
        return 20f
    }
}
